llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4278,6 +4278,26 @@ VectorizationFactor LoopVectorizationPlanner::selectVectorizationFactor() {
           if (!VPI)
             continue;
           switch (VPI->getOpcode()) {
+          // Selects are not modelled in the legacy cost model if they are
+          // inserted for reductions.
+          case Instruction::Select: {
+            VPValue *V =
+                R.getNumDefinedValues() == 1 ? R.getVPSingleValue() : nullptr;
+            if (V && V->getNumUsers() == 1) {
```

Comment:
Is a VPInstruction not always a VPSingleDefRecipe? Does this work
```suggestion
            if (VPI->getVPSingleValue()->getNumUsers() == 1) {
```